### DISCUSSION TITLE
How to dump registers

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#184] [Q&A]
I'm trying to dump the content of a register, on an IOS app.

I read in some places that into an Interceptor function there is a context object which holds the register (this.context.r0, etc) but it seems i'm not able to access it through frida-il2cpp.

Is there a way to access registers?

ty so much

**Comment by vfsfitvnm** [comment]
> Is there a way to access registers?

Nope, you need to use raw `Interceptor.replace` for that:

```ts
const callback = new NativeCallback(function (/* args */) {
    
}, "returntype", ["argtypes"]]));

Interceptor.replace(methdo.virtualAddress, callback);
```

**Reply by SamueleD78** [reply]
Thank you as usual ^^
I followed the advice.
Now i'm suppose to use the Stalker in it, but being everything in TypeScript, it's not possible for me to access dinamically created members.

For instance:

```
        Interceptor.attach(tmp_funcglob, {
            "onEnter" : function (args) {
                console.log("ONENTER func");

                var tid = Process.getCurrentThreadId();
                this.tid = tid;

                console.warn(`onEnter [ ${tid} ]`);     
                Stalker.follow(tid, {

                    transform : function (iterator) { // <----- which type that's supposed to be??

                    }

                });



            }, "onLeave" : function (retval) {
                console.log("LEAVE func");

                // cleanup
                Stalker.unfollow(this.tid);
                Stalker.garbageCollect();
            }
        });

```

the same happens for the CPU registers. Some are dinamically created into the JS in the context object, like "xN", but because of typescript they are not accessible.
